lib/pos_blockchain.go

```diff
@@ -537,6 +547,29 @@ func (bc *Blockchain) processOrphanBlockPoS(block *MsgDeSoBlock) error {
 	return errors.Wrap(err, "processBlockPoS: Problem adding block to block index: ")
 }
 
+// checkAndStoreArchivalBlock is a helper function that takes in a block and checks if it's an archival block.
+// If it is, it stores the block in the db and returns true. If it's not, it returns false, or false and an error.
+func (bc *Blockchain) checkAndStoreArchivalBlock(block *MsgDeSoBlock) (_success bool, _err error) {
+	// First, get the block hash and lookup the block index.
+	blockHash, err := block.Hash()
+	if err != nil {
+		return false, errors.Wrap(err, "checkAndStoreArchivalBlock: Problem hashing block")
+	}
+	blockNode, exists := bc.blockIndexByHash[*blockHash]
+	// If the blockNode doesn't exist, or the block is not committed, or it's already stored, then we're not dealing
+	// with an archival block. Archival blocks must have an existing blockNode, be committed, and not be stored.
+	if !exists || !blockNode.IsCommitted() || blockNode.IsStored() {
+		return false, nil
+	}
+
+	// If we get to this point, we're dealing with an archival block, so we'll attempt to store it.
+	_, err = bc.storeBlockInBlockIndex(block)
+	if err != nil {
+		return true, errors.Wrap(err, "checkAndStoreArchivalBlock: Problem storing block in block index")
```

Comment:
Shouldn't the success flag be false here?
```suggestion
		return false, errors.Wrap(err, "checkAndStoreArchivalBlock: Problem storing block in block index")
```